src/vos/vos_aggregate.c

```diff
@@ -300,6 +301,15 @@ vos_agg_obj(daos_handle_t ih, vos_iter_entry_t *entry,
 			agg_param->ap_oid = entry->ie_oid;
 			reset_agg_pos(VOS_ITER_DKEY, agg_param);
 			reset_agg_pos(VOS_ITER_AKEY, agg_param);
+			rc = oi_iter_pre_aggregate(ih);
+			if (rc < 0)
+				return rc;
+			if (rc != 0) {
+				/** We removed the key so let's reprobe */
+				*acts |= VOS_ITER_CB_YIELD | VOS_ITER_CB_DELETE;
```

Comment:
This VOS_ITER_CB_YIELD doesn't make sense to me, since we actually don't yield here. This extra yield flag could cause unnecessary re-probe when vos_iterate() iterating back to parent tree level.

Now vos_iterate() is able to detect if the iteration callback yield or not by checking the schedule sequence (see vos_iter_cb()), so callback itself doesn't need to set this yield flag anymore.

This can be fixed in follow-on patch (or when the patch needs be refreshed).

Reply:
fixed